packages/dx-grid-core/package.json

```diff
@@ -49,7 +49,8 @@
     "tslint": "^5.13.0",
     "tslint-config-airbnb": "^5.11.1",
     "typescript": "^3.3.3333",
-    "typescript-tslint-plugin": "^0.3.1"
+    "typescript-tslint-plugin": "^0.3.1",
+    "seamless-immutable": "^7.1.4"
```

Comment:
Is one more dependency (even "dev") actually required?
It seems that the following snippet would cover the case.
```js
const makeImmutable = (value) => {
  const target = value.slice();
  target.slice = () => makeImmutable(value);
  target.sort = () => {
    throw new Error('Attempt to modify!');
  };
  return target;
};
```

Reply:
In our documentation, we use the `seamless-immutable` lib and say to customers use it in your projects. I think we should test this lib without any mocks. But on the other side, we should pass any other `immutable` libraries into our controls.

Reply:
Because in the `seamless-immutable` the `reduce` method can modify the array, too. I don't know how, but tests display it.

Reply:
How does it modify array?
https://github.com/rtfeldman/seamless-immutable/blob/master/src/seamless-immutable.js#L91
https://github.com/rtfeldman/seamless-immutable/blob/master/src/seamless-immutable.js#L132